CRM/Contact/Form/Inline/Email.php

```diff
@@ -170,17 +170,9 @@ public function postProcess() {
     }
     CRM_Core_BAO_Block::create('email', $params);
 
-    // If contact has no name, set primary email as display name
-    // TODO: This should be handled in the BAO for the benefit of the api, etc.
+    // Changing email might change a contact's display_name so refresh name block content
     if (!$this->contactHasName) {
-      foreach ($params['email'] as $email) {
-        if ($email['is_primary']) {
-          CRM_Core_DAO::setFieldValue('CRM_Contact_DAO_Contact', $this->_contactId, 'display_name', $email['email']);
```

Comment:
ohh nasty - set from the form!

Reply:
![](https://media.giphy.com/media/ckvuK4mlUuohi/giphy.gif)

Reply:
lol